controllers/argocd/sso.go

```diff
@@ -33,13 +34,25 @@ const (
 )
 
 var (
-	templateAPIFound     = false
-	ssoConfigLegalStatus string
+	templateAPIFound         = false
+	deploymentConfigAPIFound = false
+	ssoConfigLegalStatus     string
 )
 
-// IsTemplateAPIAvailable returns true if the template API is present.
-func IsTemplateAPIAvailable() bool {
-	return templateAPIFound
+// CanUseKeycloakWithTemplate checks if the required APIs are available to
+// manage a Keycloak instance using Templates.
+func CanUseKeycloakWithTemplate() bool {
+	return templateAPIFound && deploymentConfigAPIFound
+}
+
+func verifyKeycloakTemplateAPIs() error {
+	found, err := argoutil.VerifyAPI(deploymentConfig.GroupVersion.Group, deploymentConfig.GroupVersion.Version)
+	if err != nil {
+		return err
+	}
+	deploymentConfigAPIFound = found
+
+	return verifyTemplateAPI()
```

Comment:
nit: I recommend moving the contents of verifyTemplateAPI into this function.  'verifyKeycloakTemplateAPIs' is the only function calling it, and we probably want to make sure that both variables are always updated at the same time.